zetaclient/chains/evm/observer/inbound.go

```diff
@@ -145,35 +145,40 @@ func (ob *Observer) ProcessInboundTrackers(ctx context.Context) error {
 		}
 		ob.Logger().Inbound.Info().Msgf("checking tracker for inbound %s chain %d", tracker.TxHash, ob.Chain().ChainId)
 
-		// if the transaction is sent to the gateway, this is a v2 inbound
+		// try processing the tracker for v2 inbound
 		gatewayAddr, gateway, err := ob.GetGatewayContract()
 		if err != nil {
 			ob.Logger().Inbound.Debug().Err(err).Msg("error getting gateway contract for processing inbound tracker")
 		}
-		if err == nil && tx != nil && ethcommon.HexToAddress(tx.To) == gatewayAddr {
-			if err := ob.ProcessInboundTrackerV2(ctx, gateway, tx, receipt); err != nil {
+		if err == nil && tx != nil {
```

Comment:
Can `tx` be nil? If no, let's remove this check, otherwise `tx == nil` case should be handled above

Reply:
https://github.com/zeta-chain/node/pull/3226/commits/c00f9644a73f30771a74adeaa15fd9a6ffbdef40